### PR TITLE
Add support for multiple databases for Active Storage/Action Text/Action Mailbox

### LINKED ISSUE #38755: Active Storage and multiple databases
Is is possible to use active storage with multi databases in rails? I have the following models:

### Steps to reproduce

```
class ApplicationRecord < ActiveRecord::Base
  self.abstract_class = true
end


class OtherRecord < ActiveRecord::Base
  self.abstract_class = true

  connects_to database: { writing: :other_db }
end


class SomeModel < OtherRecord
  has_many_attached :files
end
```

I've added active storage migrations scoped to the secondary database and necessary tables were created there but ActiveStorage seems to look for tables in primary database. Have anyone a solution for that?

### Expected behavior
I should have an option to point active storage to use secondary database.

### Actual behavior
ActiveStorage seems to look for tables in primary database.

### System configuration
**Rails version**:
gem 'rails', '~> 6.0.2.1'

**Ruby version**:
ruby-2.7

### PR DESCRIPTION
**Pull request by n-studio**
Fixes: https://github.com/rails/rails/issues/38755

Similar to how Solid Queue and Solid Cache are configured:
https://github.com/rails/solid_queue/blob/main/app/models/solid_queue/record.rb
https://github.com/rails/solid_cache/blob/main/app/models/solid_cache/record.rb

### CODE REVIEW

**Comment by zzak** [comment]
This should be done as a class variable and configured in the engine during initialization. I can dig up some examples if you need more pointers.

**Comment by n-studio** [comment]
@zzak Yes, please!

**Comment by n-studio** [comment]
@zzak I updated the PR to use class variables.

I tried to configure the tests to run multiple databases but I keep running into `ActiveRecord::StatementTimeout: SQLite3::BusyException: database is locked` errors.

I'll try to figure out why when I can.

File: actiontext/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ## Rails 8.0.0.beta1 (September 26, 2024) ##
 
+*   Add support for multiple databases for Action Text.
+
+    `config.action_text.connects_to = { database: { writing: :primary, reading: :primary_replica } }`
+
+    *Matthew Nguyen*
+
```
Comment (zzak) [inline actiontext/CHANGELOG.md:8]: This should go at the top of the file

**Comment by zzak** [comment]
> I tried to configure the tests to run multiple databases but I keep running into `ActiveRecord::StatementTimeout: SQLite3::BusyException: database is locked` errors.

Maybe this is related: https://github.com/rails/rails/blob/main/activestorage/test/database/setup.rb

There is a dummy app that is used for running the tests:
https://github.com/rails/rails/tree/main/activestorage/test/dummy

**Comment by n-studio** [comment]
@zzak I did try to configure the dummy. Sorry for the lack of explanation, let me write a full comment with the details on the failing tests!

**Comment by rafaelfranca** [comment]
See https://github.com/rails/rails/pull/53371#issuecomment-2423264283

**Pull request closed without merging**